api/src/middleware/assessmentsRouter.ts

```diff
@@ -36,9 +37,47 @@ assessmentsRouter.get('/', async (req, res, next) => {
 assessmentsRouter.get(
   '/curriculum/:curriculumAssessmentId',
   async (req, res, next) => {
-    res.json();
+    const { curriculumAssessmentId } = req.params;
+    const curriculumAssessmentIdParsed = Number(curriculumAssessmentId);
+    if (
+      !Number.isInteger(curriculumAssessmentIdParsed) ||
+      curriculumAssessmentIdParsed < 1
+    ) {
+      next(
+        new BadRequestError(
+          `"${curriculumAssessmentIdParsed}" is not a valid submission ID.`
+        )
+      );
+      return;
+    }
+
+    const { principalId } = req.session;
+    const programId = await getProgramIdByProgramAssessmentId(
+      curriculumAssessmentIdParsed
+    );
+    const programIdParsed = Number(programId);
+    const programRole = await getPrincipalProgramRole(
+      principalId,
+      programIdParsed
+    );
```

Comment:
I... think? I'm following your logic here, but the fatal flaw here is that you can't directly get a program ID given a curriculum assessment ID.

Instead, you'll want to check L143 through L161 to see how we tackled this logic with Paryssa's route.